### ISSUE TITLE
Phoneme-blending guessing-floor policy covers 4 of 13 B-outcome models

### ISSUE THREAD

**Issue by frankbuckley** [dseinternational/language-reading-predictors#608]
> [!NOTE]
> Drafted by a LLM-based AI tool (Claude Code/Opus 5).
>
> Coverage table, counts and decision status updated 2026-08-24 by a LLM-based AI tool (Claude Code/Opus 5) — see the decision comment below and PR #616.

## Summary

`METHODS.md` requires that **any** headline phoneme-blending (`B`) interpretation be accompanied by the mechanically motivated guessing-floor link sensitivity `mu = 1/3 + (2/3) * inverse_logit(eta)`. Fourteen registered models fit `outcome_symbol="B"`. Only **three pairs** are built, and the release gate reaches only the **ITT**, **level-factors** and **waitlist-crossover** families — so eight models across five families publish an unqualified blending result that the stated policy says is not releasable. Two of the three built pairs bind nothing today, because their companions have never been fitted at reporting tier.

This surfaced while remediating #587, whose audit raised it against `lrp-rli-dose-084`. It is not a dose-family problem: `dose-084` is one of eight in the same position, and fixing it alone would apply the policy to one eighth of the affected fits.

## Current coverage

| Model | Kind | Link | Companion | Release gate reaches it | Reporting fit stored |
| --- | --- | --- | --- | --- | --- |
| `lrp-rli-itt-008` | `itt` | logit | `itt-108` | yes — `BLENDING_LINK_MODELS` | yes |
| `lrp-rli-itt-108` | `itt` | guessing floor | `itt-008` | yes — `BLENDING_LINK_MODELS` | yes |
| `lrp-rli-lf-006` | `level_factors` | logit | `lf-106` | yes — `_level_blending_pair_release_failures` (#584 decision 2) | yes |
| `lrp-rli-lf-106` | `level_factors` | guessing floor | `lf-006` | yes — same | **no** |
| `lrp-rli-did-003` | `did` | logit | `did-103` | yes — `_did_blending_pair_release_failures` (#576 finding 2) | yes |
| `lrp-rli-did-103` | `did` | guessing floor | `did-003` | yes — same | **no** |
| `lrp-rli-al-006` | `aligned` | logit | **none** | **no** | yes |
| `lrp-rli-ca-007` | `concurrent` | logit | **none** | **no** | yes |
| `lrp-rli-dose-084` | `dose_response` | logit | **none** | **no** | yes |
| `lrp-rli-gf-006` | `gain_factors` | logit | **none** | **no** | yes |
| `lrp-rli-gf-106` | `gain_factors` | logit | **none** | **no** | yes |
| `lrp-rli-gf-206` | `gain_factors` | logit | **none** | **no** | yes |
| `lrp-rli-lf-206` | `level_factors` | logit | **none** | **no** — see note below | **no** |
| `lrp-rli-med-087` | `mediation` | logit | **none** | **no** | yes |

`lrp-rli-lf-206` resolves the open question in the original table: it routes to the level checker by `kind`, but `evaluate_level_blending_link_pair` requires either membership of the registered `lf-006`/`lf-106` pair or a plan declaring `link_sensitivity_required_for_release`. It has neither, so the checker returns `required: False` and it publishes unpaired.

The gate's shape is in `release.py`'s `_blending_pair_release_failures`: it routes `kind == "level_factors"` to the level checker and `kind == "did"` to the DiD checker, handles `kind == "itt"` via `BLENDING_LINK_MODELS` plus the plan's `link_sensitivity_required_for_release`, and returns `()` — no failures — for **every other kind**. `blending_sensitivity.evaluate_local_blending_link_sensitivity` likewise short-circuits with `{"required": False, "ready": True, "reason": "not a B-link fit"}` for any model outside `BLENDING_LINK_MODELS`, which is what `docs/models/_partials/_key_findings.qmd` consults at render time for every `outcome_symbol == "B"` fit. So those eight fits set `_kf_blending_required = True` and then publish anyway.

Reproduce the coverage table with:

```bash
uv run python -c "
import pathlib, re
from language_reading_predictors.statistical_models.blending_sensitivity import (
    BLENDING_PRIMARY_MODEL_ID as I_P, BLENDING_COMPANION_MODEL_ID as I_C)
from language_reading_predictors.statistical_models.level_factors import (
    LEVEL_BLENDING_PRIMARY_MODEL_ID as L_P, LEVEL_BLENDING_COMPANION_MODEL_ID as L_C)
from language_reading_predictors.statistical_models.did import (
    DID_BLENDING_PRIMARY_MODEL_ID as D_P, DID_BLENDING_COMPANION_MODEL_ID as D_C)
PAIRS = {I_P: I_C, I_C: I_P, L_P: L_C, L_C: L_P, D_P: D_C, D_C: D_P}
OUT = pathlib.Path('output/statistical_models/models')
for p in sorted(pathlib.Path('src/language_reading_predictors/statistical_models').glob('lrp_*.py')):
    s = p.read_text()
    if 'outcome_symbol=\"B\"' not in s:
        continue
    mid = re.search(r'model_id=\"([^\"]+)\"', s).group(1)
    kind = re.search(r'kind=\"([^\"]+)\"', s).group(1)
    link = 'guessing_floor' if 'three_choice_guessing_floor' in s else 'logit'
    print(f'{mid:22s} {kind:16s} {link:16s} paired={str(mid in PAIRS):5s} fit={(OUT / (mid + \"-reporting\")).is_dir()}')
"
```

## Why it matters

Phoneme blending has ten three-alternative forced-choice items, so a child responding at random scores about 3.3 out of 10. The ordinary Beta-Binomial inverse-logit mean does not know that and permits expected scores anywhere in (0, 1). This is not hypothetical: `lrp_rli_lf_106.py`'s docstring records that the fitted `lf-006` posterior used that room — 24 of 215 rows with posterior-mean expected proportions below one third, and at timepoint 2 it was 8 of 54 rows and 16% of the posterior mass. There is no reason to expect the other nine fits to behave differently, and none of them has been checked.

The policy is also currently applied unevenly rather than deliberately, which is the worse failure mode: a reader has no way to tell that `itt-008`'s blending result carries a validated link pair while `gf-006`'s identical-looking blending result does not.

## Two things to decide — **both settled 2026-08-24** (see the decision comment below; PR #616)

**1. Scope of the policy — it stands as written.** ~~Either it means what it says ... or it was meant to cover randomised headline effects only.~~ **Settled: the policy binds every registered model with `outcome_symbol="B"`, in every family**, whether the published quantity is a randomised contrast or an adjusted association. The gate is keyed on the outcome symbol rather than on `kind`.

Two measurements decided it rather than the argument from first principles. The defect is *realised* in every stored `B` fit, not merely permitted — below-chance posterior mass runs 4.9 % to 13.7 %, with rows in every fit where the model is 80–100 % certain of below-chance performance — so a threshold rule would exempt nothing. And the link changes the finding rather than rescaling it: at reporting tier the ITT pair moves from 0.99 items (89 % CI 0.22 to 1.74, evidence **strong**) to 0.49 items (89 % CI −0.14 to 1.09, evidence **suggestive**), the practical verdict inverting from P(negligible) 0.51 to 0.91, while the floor link fits marginally **better** by PSIS-LOO.

The `dose-084` argument — that the floor may bear differently on an observational slope — fails on its own example: `METHODS.md` defines every dose fit's focal estimand as *the natural-scale treated-row dose marginal*, which is link-dependent exactly as an items-scale treatment effect is.

**2. Evidence tier per family — one tier.** ~~A third tier for the remaining families is probably acceptable.~~ **Settled: every pair is bound by the content-addressed archive**, not only the ITT one; the local two-directory check is retired as a permanent arrangement rather than promoted to a second tier. A reader cannot tell from a rendered report which apparatus backs a given card, so tiering by family would preserve exactly the ambiguity this issue exists to remove.

## Proposed work

- [x] Settle decision 1 above and record it in `notes/`; update `METHODS.md` to say exactly what it means. — done 2026-08-24, PR #616 (`notes/202608242000-blending-guessing-floor-scope-608.md`; decision 2 settled at the same time).
- [ ] The policy stands as written, so: add `score_mean_link` support to the families that lack it (`aligned`, `concurrent`, `dose_response`, `gain_factors`, `mediation` — `itt`, `level_factors` and `did` already have it), register a guessing-floor companion per affected primary, and fit both at reporting tier. **Check the run-plan digest first**: adding the setting changes the resolved plan, and pairing requires both halves to share a source commit and plan, so the eight existing primaries may need refitting too — the difference between eight reporting fits and sixteen.
- [ ] Generalise the release check so it is keyed on `outcome_symbol == "B"` rather than on `kind`, so a **new** B-outcome model in any family fails closed instead of silently publishing unpaired. Per decision 2 there is one evidence tier, so this is a single archive-backed path rather than a per-family tier declaration; the archive needs parameterising by pair and by per-family focal columns (it is currently hardcoded to one global pair with ITT-shaped free variables and summary-column map).
- [ ] Add a registry-derived test that every `outcome_symbol="B"` model is either paired or explicitly exempted, so this cannot regrow.
- [ ] Compute the gain family's treatment marginal as a **score mean** rather than a raw inverse-logit, and map each family's empirical-Bayes intercept anchor back through the link — both follow from the decision rather than being discretionary, and the level family's anchor moved 1.1 logits.
- [ ] Fit the two registered-but-never-fitted companions, `lrp-rli-lf-106` and `lrp-rli-did-103`, plus `lrp-rli-lf-206` which has no stored fit at all. **Confirmed 2026-08-24**: neither companion has a reporting-tier fit, so two of the three built pairs bind nothing today. Re-run `release.evaluate_publication` over `lrp-rli-lf-006`'s stored directory as part of this — its plan records `link_sensitivity_required_for_release=False` (it predates #584 decision 2) while `release_decision.json` says `ok`.

## Interim state

`lrp-rli-dose-084` carries a standing "qualified result" sentence in its key findings as of #606, stating that the required companion has not been built for its family and that the blending association should be read as provisional. The other seven uncovered fits carry no such qualification, and neither do the two built-but-unfitted pairs whose companions do not yet exist.

## Non-goals

- Changing the guessing-floor link itself, or the `1/3 + (2/3) * inverse_logit(eta)` form.
- Revisiting the ITT pair's content-addressed archive apparatus, which works.

**Comment by frankbuckley** [comment]
> [!NOTE]
> Drafted by a LLM-based AI tool (Claude Code/Opus 5).

## Decision (2026-08-24)

Both questions in "Two things to decide" are settled. Recorded in `METHODS.md`, the three agent-instruction files and `notes/202608242000-blending-guessing-floor-scope-608.md`; PR #616.

**1. Scope — the policy stands as written.** The guessing-floor link sensitivity binds **every registered model with `outcome_symbol="B"`, in every family**, whether the published quantity is a randomised contrast or an adjusted association. The release gate keys on the outcome symbol rather than on `kind`, so a new `B` model in any family fails closed until it is paired or carries a recorded, dated exemption.

**2. Evidence tier — one tier.** Every pair is bound by the content-addressed archive, not only the ITT one. The local two-directory pair check is retired as a permanent arrangement rather than promoted to a second tier.

## What settled it

Two measurements, rather than an argument from first principles. Both are reproducible from stored artefacts without a refit.

**The defect is realised in every stored `B` fit, not merely permitted.** For a Beta-Binomial with a logit score mean this is directly measurable — `expit(eta) < 1/3` is `eta < -log 2`, and `eta` is stored as a Deterministic in every trace:

| Model | Family | Rows with posterior-mean expected proportion < ⅓ | Posterior mass below ⅓ | Worst single row |
| --- | --- | --- | --- | --- |
| `lf-006` | `level_factors` | 24/215 | 13.7 % | 99.5 % |
| `did-003` | `did` | 15/162 | 13.1 % | 99.6 % |
| `med-087` | `mediation` | 5/53 | 11.6 % | 99.1 % |
| `gf-006` | `gain_factors` | 15/161 | 10.7 % | 99.8 % |
| `gf-206` | `gain_factors` | 13/161 | 10.5 % | 99.9 % |
| `ca-007` | `concurrent` | 4/54 | 9.7 % | 80.4 % |
| `itt-008` | `itt` | 5/54 | 8.9 % | 100 % |
| `gf-106` | `gain_factors` | 4/135 | 8.4 % | 91.5 % |
| `dose-084` | `dose_response` | 8/160 | 7.0 % | 100 % |
| `al-006` | `aligned` | 2/54 | 4.9 % | 98.0 % |

The `lf-006` row reproduces the 24/215 already recorded in `lrp_rli_lf_106.py`'s docstring, which validates the method against an independently computed number. **No fit is clean**: every one contains rows where the model is 80–100 % certain a child performs below chance on a three-alternative forced-choice test, and the mildest still carries 55 % of `itt-008`'s below-chance mass.

This also disposes of the obvious middle path. A threshold rule — require the pair only where the floor demonstrably binds — exempts **nothing** on current data, so it is the strict policy plus a threshold to defend. It is still worth implementing as the fail-closed gate mechanism for future models, but it is not a scope rule.

**The link changes the finding rather than rescaling it.** `itt-008` / `itt-108` is the only pair with both links fitted at reporting tier, on identical rows with identical data and sampling contracts:

| | `itt-008` (logit) | `itt-108` (guessing floor) |
| --- | --- | --- |
| Effect, items | 0.99 | **0.49** |
| 89 % CI, items | 0.22 to 1.74 | **−0.14 to 1.09** |
| P(effect > 0) | 0.980 | 0.893 |
| Evidence label | **strong** | **suggestive** |
| P(meaningful benefit) | 0.49 | **0.088** |
| P(practically negligible) | 0.51 | **0.91** |

The estimate halves, the interval crosses zero, the evidence label drops a rung and the practical verdict inverts — while `guessing_floor_minus_logit_elpd = +1.09` (SE 2.68), so the floor link fits marginally **better** by PSIS-LOO. A constraint that costs nothing in fit and is mechanically motivated by the test's own design makes the ordinary-link number the one requiring justification, not the floored one.

## On the `dose-084` argument specifically

The issue asks whether the guessing floor "bears differently on a slope than on a headline contrast", and notes that `dose-084`'s quantity is an observational attendance slope. That argument does not survive its own example: `METHODS.md` defines every dose fit's focal estimand as *the natural-scale treated-row dose marginal*. The link determines the mapping from the latent scale to the reported one, so any natural-scale quantity is link-dependent exactly as an items-scale treatment effect is, regardless of what identifies it.

A latent-logit coefficient's sign is more robust to the link than a natural-scale marginal is. That is an argument for reporting latent-scale quantities where they answer the question — not for leaving a misspecified link under a natural-scale headline.

## Three corrections to the issue text

The coverage table above has been updated in the issue body; the changes are:

1. **It is 8 uncovered models, not 9, and 14 `B` models, not 13.** `did-003` / `did-103` were paired by #613 after this issue was filed.
2. **`lf-206`'s "see note below" resolves to *not gated*.** It routes to the level checker by `kind`, but `evaluate_level_blending_link_pair` requires either membership of the registered `lf-006`/`lf-106` pair or a plan declaring `link_sensitivity_required_for_release`. `lf-206` has neither, so the checker returns `required: False` and it publishes unpaired.
3. **Two of the three named pairs bind nothing today.** `lf-106` and `did-103` are registered with the code in place but have **never been fitted at reporting tier**, and `lf-206` has no stored fit at all. Those fits are prerequisites for the implementation batch, not extras — this confirms the last bullet of "Proposed work".

## One thing to check before scheduling compute

Adding `score_mean_link` to a family's settings changes the resolved run plan, and pairing requires both halves to share a source commit and run plan. Even where a fitted primary is numerically unchanged its provenance may no longer match, so the run-plan digest should be checked before assuming the eight existing primaries stand. It is the difference between eight reporting fits and sixteen.

## Two implementation items that follow from the decision

Neither is discretionary, and both are easy to miss:

- The gain family's treatment marginal must be computed as a **score mean** rather than a raw inverse-logit, or a floor-link posterior will publish an ordinary-link number. The level family's equivalent would have done exactly that had it not been caught.
- Each family with an empirical-Bayes intercept anchor must map it back through the link, as the level family's did — it moved 1.1 logits.

## Deliberately left open

The threshold and reporting format for a per-fit below-chance diagnostic alongside each `B` card, so a reader can see how hard the floor binds in that fit. Recommended, but a reporting-design question rather than a scope one, and it should not hold up the implementation.
